### ISSUE TITLE
build: make `publish-artifacts` _the_ program used to publish S3 artifacts

### ISSUE THREAD

**Issue by jlinder** [cockroachdb/cockroach#53219]
With the new release process, we are using the same build and publish process for both internal testing builds and publishing release builds (just to different buckets). This removes the need for separate `publish-artifacts` and `publish-provisional-artifacts`.

Some functionality from `publish-provisional-artifacts` needs to be retained. Specifically, the ability to publish the `latest` artifacts and to set the build tag used in the published binaries. Once that functionality is folded into `publish-artifacts` (and we are transitioned to using the new release process), `publish-provisional-artifacts` can be removed.

We also want to publish `latest-RELEASE_BRANCH` artifacts, which is a straight-forward extension to publishing the `latest` artifacts.

The tasks:

- [ ] add ability to set the build tag in published binaries
- [ ] add ability to publish the `latest` artifacts to `publish-artifacts`
- [ ] make the `latest` artifacts use the [webpage redirect functionality of S3](https://docs.aws.amazon.com/AmazonS3/latest/dev/how-to-page-redirect.html) for buckets that support this feature
- [ ] add ability to publish the `latest-RELEASE_BRANCH` artifacts
- [ ] update `build/release/teamcity-make-and-publish-build.sh` and `build/release/teamcity-publish-release.sh` to use `publish-artifacts` for all S3 artifact publishing
- [ ] remove `publish-provisional-artifacts` (once the above work is done and we are using the new release process)
- [ ] document the CLI options and what how they affect the output

Epic RE-227
Jira issue: CRDB-3886

**Comment by rail** [comment]
Most of the above are not applicable anymore or already implemented. Closing.
